test/e2e/tests/test_endpoint.py

```diff
@@ -410,16 +391,21 @@ def delete_endpoint_test(self, sagemaker_client, xgboost_endpoint):
         endpoint_name = resource["spec"].get("endpointName", None)
 
         _, deleted = k8s.delete_custom_resource(reference)
-        assert deleted is True
+        assert deleted
 
         # resource is removed from management from controller side if call to deleteEndpoint succeeds.
         # Sagemaker also removes a 'Deleting' endpoint pretty quickly, but there might be a lag
-        # If we see errors in this part of test, can add a loop in future or consider changing controller
-        # to wait for SageMaker
-        time.sleep(10)
-        assert (
-            self._describe_sagemaker_endpoint(sagemaker_client, endpoint_name) is None
-        )
+        # We wait maximum of 30 seconds for this clean up to happen
+        endpoint_deleted = False
+        for _ in range(6):
+            time.sleep(5)
```

Comment:
i think its more reasonable to do `range(3)` and `sleep(10)`

Reply:
fair point, could reduce the checks by half with an at max lag extra lag of 5seconds.

Reply:
done